--- jp2.c (4517)
+++ jp2.c (working copy)
@@ -56,7 +56,8 @@
 static void jp2_write_ihdr(opj_jp2_t *jp2, opj_cio_t *cio);
 static void jp2_write_bpcc(opj_jp2_t *jp2, opj_cio_t *cio);
 static opj_bool jp2_read_bpcc(opj_jp2_t *jp2, opj_cio_t *cio);
-static void jp2_write_colr(opj_jp2_t *jp2, opj_cio_t *cio);
+extern void jp2_write_colr(opj_jp2_t *jp2, opj_cio_t *cio);
+extern void jp2_write_xml(opj_jp2_t *jp2, opj_cio_t *cio);
 /**
 Write the FTYP box - File type box
 @param jp2 JP2 handle
@@ -297,7 +298,7 @@
 	return OPJ_TRUE;
 }
 
-static void jp2_write_colr(opj_jp2_t *jp2, opj_cio_t *cio) {
+static void jp2_write_colr_orig(opj_jp2_t *jp2, opj_cio_t *cio) {
 	opj_jp2_box_t box;
 
 	box.init_pos = cio_tell(cio);
@@ -1176,6 +1177,7 @@
 	jp2_write_ftyp(jp2, cio);
 	/* JP2 Header box */
 	jp2_write_jp2h(jp2, cio);
+	jp2_write_xml(jp2, cio);
 
 	if( jp2->jpip_on){
 	  pos_iptr = cio_tell( cio);
