diff -ur cfitsio/putkey.c cfitsio-3.47/putkey.c
--- cfitsio/putkey.c	2019-05-08 17:34:50.000000000 +0200
+++ cfitsio-3.47/putkey.c	2019-10-10 23:17:06.000000000 +0200
@@ -2413,12 +2413,12 @@
         }
 
       /* write standard block of self-documentating comments */
-      ffprec(fptr,
+/*    ffprec(fptr,
       "COMMENT   FITS (Flexible Image Transport System) format is defined in 'Astronomy",
       status);
       ffprec(fptr,
       "COMMENT   and Astrophysics', volume 376, page 359; bibcode: 2001A&A...376..359H",
-      status);
+      status); */
     }
 
     else  /* an IMAGE extension */
