--- cfitsio/putkey.c	2018-05-09 21:16:01.000000000 +0200
+++ cfitsio3450/putkey.c	2018-05-17 10:37:19.000000000 +0200
@@ -2413,12 +2413,12 @@
         }
 
       /* write standard block of self-documentating comments */
-      ffprec(fptr,
+/*    ffprec(fptr,
       "COMMENT   FITS (Flexible Image Transport System) format is defined in 'Astronomy",
       status);
       ffprec(fptr,
       "COMMENT   and Astrophysics', volume 376, page 359; bibcode: 2001A&A...376..359H",
-      status);
+      status); */
     }
 
     else  /* an IMAGE extension */
